dev/com.ibm.ws.jca_fat_bvt.jms/fat/src/test/jca/jms/example/tests/JCAStoreSampleAppTest.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package test.jca.jms.example.tests;
+
+import java.io.File;
+import java.util.List;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import test.jca.jms.example.fat.FATSuite;
+
+@RunWith(FATRunner.class)
+public class JCAStoreSampleAppTest extends FATServletClient {
+    private static final String WAR_NAME = "jcastore";
+
+    @Server(FATSuite.SERVER)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Build the web module and application
+        WebArchive fvtweb_war = ShrinkWrap.create(WebArchive.class, WAR_NAME + ".war");
+        fvtweb_war.addPackage("test.jca.jms.example.mdb");
+        fvtweb_war.addPackage("test.jca.jms.example.web");
+        fvtweb_war.addAsWebInfResource(new File("test-applications/" + WAR_NAME + "/resources/WEB-INF/web.xml"));
+        ShrinkHelper.exportToServer(server, "apps", fvtweb_war);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    public void testJCAStoreSampleApp() throws Exception {
+        runTest(server, WAR_NAME + "/JCAStoreServlet", "purchase10Lightbulbs");
+        runTest(server, WAR_NAME + "/JCAStoreServlet", "purchase12Lightbulbs");
+        // MDB should run now and restock the amount to 35 - need to wait for that to happen
+        boolean found = false;
+        for (int i = 0; i < 60 && !found; i++) {
+            List<String> lines = server.findStringsInFileInLibertyServerRoot("InventoryTrackerMDB completed processing for item", "logs/messages.log");
+            if (!lines.isEmpty()) {
+                found = true;
+            }
+
+            if (!found)
+                Thread.sleep(500);
+        }
+        if (!found)
+            throw new Exception("Didn't find output from MDB in messages.log");
```

Comment:
Instead of polling every 500ms here, we should update this to use the `waitForStringInLogs` method.